MyXamarinApp.API/Repository/EmployeeRepository.cs

```diff
@@ -39,17 +74,31 @@ public async Task<EmployeeModel> AddEmployee(EmployeeModel employeeModel)
             var employee = _mapper.Map<Employee>(employeeModel);
             _context.Employees.Add(employee);
             await _context.SaveChangesAsync();
-            return employeeModel;
+            return await GetEmployeeById(employee.EmployeeGuid);
+
+        }
+
+        /// <summary>
+        /// Updates the employee details to the Employee table.
+        /// </summary>
+        /// <param name="employeeModel">Employee to be added.</param>
+        /// <returns>Employee model of the new Employee.</returns>
+        public async Task<EmployeeModel> UpdateEmployee(EmployeeModel employeeModel)
+        {
+            var employee = _mapper.Map<Employee>(employeeModel);
+            _context.Employees.Update(employee);
+            await _context.SaveChangesAsync();
+            return _mapper.Map<EmployeeModel>(employee);
         }
 
         /// <summary>
         /// Removes the employee from the Employee table.
         /// </summary>
         /// <param name="employeeId">Employee ID of the employee to be removed.</param>
         /// <returns>True if employee is removed.</returns>
-        public async Task<bool> RemoveEmployee(int employeeId)
+        public async Task<bool> RemoveEmployee(Guid employeeId)
         {
-            var employee = _context.Employees.Where(x => x.EmpId.Equals(employeeId)).FirstOrDefault();
+            var employee = _context.Employees.Where(x => x.EmployeeGuid.Equals(employeeId)).FirstOrDefault();
```

Comment:
This we are doing on the controller side too and we can avoid an overhead on the api by simply passing the employee object here and do  _context.Employees.Remove(employee);